dspace-api/src/main/java/org/dspace/core/HibernateDBConnection.java

```diff
@@ -164,6 +164,83 @@ private void configureDatabaseMode() throws SQLException {
      */
     @Override
     public <E extends ReloadableEntity> void uncacheEntity(E entity) throws SQLException {
-        getSession().evict(entity);        
+        if(entity != null) {
+            if (entity instanceof DSpaceObject) {
+                DSpaceObject dso = (DSpaceObject) entity;
+
+                // The metadatavalue relation has CascadeType.ALL, so they are evicted automatically
+                // and we don' need to uncache the values explicitly.
+
+                if(Hibernate.isInitialized(dso.getHandles())) {
```

Comment:
WorkSpace and WorkflowItems do not have any handles. May this lead to a problem?